### ISSUE TITLE
AGTQRTSVStreamer parsing tab char Bug(?)

### ISSUE THREAD

**Issue by wThomas84** [franzinc/agraph-java-client#3]
Dear Franz Inc,

we encountered a problem while reading a String that contains a tab character on reading from Allegrograph 6.4.6 a SPARQL query over TupleQueryResult, parsed over AGTQRTSVStreamer.

We use the agraph-java-client version that is published over Maven Central in version 2.2.1.

In order to reproduce the malicious behaviour, we implemented a failing test that can be found in our forked repository:
https://github.com/as-ideas/agraph-java-client/blob/12db7abbeb5b40c0b181c06a6e759f718a7729e7/src/test/java/test/RepositoryConnectionTest.java#L641

The git commit the test is implemented on, is related to the version that is deployed on maven central (hopefully).

Do you have any suggestions how we could circumvent this behaviour?

Thanks + Best Regards
Tom + Julian (Ideas Engineering)

**Comment by Riptidemyth** [comment]
Hello Tom, Julian,

AGTQRTSVStreamer parser treats the tab character as separator while parsing, leading to the parsing error we assume you get. Since the default response format tuple queries is TSV (Tab Separated Values), the AGTQRTSVStreamer parser counts on tabs not being anywhere in the text, since tabs have special meaning there.

One way to avoid it would be to configure the response format and change it to something like JSON or SPARQL (the latter is plain XML). You can do this by typing something like this:

`((AGRepositoryConnection)testCon).prepareHttpRepoClient().setPreferredTQRFormat(TupleQueryResultFormat.JSON);`

After that the parsing error went away, however the assert below fails, but for a different reason. In order to compare `nameTrudy` literal with `bs.getValue("name").stringValue()` one has to convert Literal to plain string by calling `getLabel()` on it. The resulting assert would look something like this:

`assertEquals("", nameTrudy.getLabel(), bs.getValue("name").stringValue());`

With these modifications the test passes for me.

Please let me know if that helps and addresses your problem.

Best regards,
Bogdan

**Comment by tsznuk** [comment]
It might be worth noting that the SPARQL TSV format [specification](https://www.w3.org/TR/sparql11-results-csv-tsv/#tsv-terms) requires tab characters to be escaped by the server, so this seems to be a bug in AllegroGraph, not in the Java client.

**Comment by wThomas84** [comment]
Hello,

thank you very much for your answers. 

Regarding the test assertion, yes, true it was wrong and we fixed the assertion.

In order to mitigate the wrong behaviour we indeed switched to another `AGTQRStreamer` - the `AGTQRXMLStreamer`.

We also tried the `AGTQRJSONStreamer` but encountered incorrect result deserialisation, too.
While the response stream from the database contains whole, valid json (debugger-watched in `AGTQRJSONStreamer:55`), the resulting TupleQueryResult contained only one `BindingSet`.
Our assumption is that the JSON processing is falsely aborted too early somehow.

Best
/Tom
